receiver/receivercreator/README.md

```diff
@@ -19,6 +21,10 @@ endpoints that you may be interested in. The configured rules will be
 evaluated for each endpoint discovered. If the rule evaluates to true then
 the receiver for that rule will be started against the matched endpoint.
 
+If you use the receiver creator in multiple pipelines of differing telemetry types,
+but a given dynamically instantiated receiver doesn't support one of the pipeline's type,
+it will effectively lead to a logged no-op that won't cause a collector service failure.
```

Comment:
@dmitryax Could you provide an example of this?

Reply:
```suggestion
> Note: When using receiver creator in multiple pipelines with different telemetry types, the Collector might crash if one of the receivers doesn't support a pipeline type. Make sure the instantiated receivers can support the pipeline type.
```

Reply:
@rmfitzpatrick is probably better placed to comment, but I think you misunderstood what is documented there. Anyway, happy to clarify in a separate issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23014